webapp/cas/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/gin-gonic/gin"
+)
+
+const BASE_URL = "http://localhost:4200"
+
+func getRouter(cas CAS) *gin.Engine {
+	redirectWithPath := func(getPath func(*gin.Context) string) func(*gin.Context) {
+		return func(c *gin.Context) {
+			resp, err := http.Get(BASE_URL + getPath(c))
+			if err != nil {
+				c.Error(err)
+				c.Abort()
+			}
+			defer resp.Body.Close()
+
+			c.DataFromReader(
+				resp.StatusCode,
+				resp.ContentLength,
+				resp.Header.Get("Content-Type"),
+				resp.Body,
+				map[string]string{},
+			)
+		}
+	}
+	redirectToRoot := redirectWithPath(func(*gin.Context) string { return "/" })
+	redirect := redirectWithPath(func(c *gin.Context) string { return c.Request.URL.String() })
+
+	r := gin.Default()
+
+	r.GET("/api/v0/cas/p3/serviceValidate", func(c *gin.Context) {
+		c.XML(http.StatusOK, cas.ServiceValidateXML(
+			c.Query("service"), c.Query("ticket"),
+		))
+	})
+
+	r.GET("/", redirectToRoot)
+	r.GET("/register/*path", redirectToRoot)
```

Comment:
That feels awkward - if we add a new path in the webapp, we need to add it also in the go? Can't you have a 'catch-all', and then prioritise?

Reply:
Sadly, no, as gin is *very* fast, it uses a single Trie without fallback. I might be able to work it around by adding a preprocessor to every request, if you need